extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSources.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * @since 4.1
+ */
+public final class ElasticSources {
+
+    private static final int DEFAULT_PORT = 9200;
+
+    private ElasticSources() {
+    }
+
+    /**
+     * Creates a source which queries local instance of Elasticsearch for all documents
+     * <p>
+     * Useful for quick prototyping. See other methods {@link #elastic(SupplierEx, SupplierEx, FunctionEx)}
+     * and {@link #builder()}
+     */
+    public static BatchSource<String> elastic() {
+        return elastic(() -> new RestHighLevelClient(
+                RestClient.builder(new HttpHost("localhost", DEFAULT_PORT))
+        ));
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Queries all indexes for all documents.
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    public static BatchSource<String> elastic(@Nonnull SupplierEx<RestHighLevelClient> clientSupplier) {
+        return elastic(clientSupplier, SearchHit::getSourceAsString);
+    }
+
+    /**
+     * Creates a source which queries local instance of Elasticsearch for all documents
+     * Uses {@link SearchHit#getSourceAsString()} as mapping function
+     */
+    public static <T> BatchSource<T> elastic(@Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(() -> new RestHighLevelClient(
+                        RestClient.builder(new HttpHost("localhost", DEFAULT_PORT))
+                ),
+                mapHitFn
+        );
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     * Uses provided mapHitFn to map results.
+     * Queries all indexes for all documents.
+     *
+     * @param clientSupplier RestHighLevelClient supplier
+     * @param mapHitFn       supplier of a function mapping the result from SearchHit to a result type
+     * @param <T>            result type returned by the map function
+     */
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull FunctionEx<? super SearchHit, T> mapHitFn) {
+        return elastic(clientSupplier, SearchRequest::new, mapHitFn);
+    }
+
+    /**
+     * Creates a source which queries Elasticsearch using client obtained from {@link RestHighLevelClient} supplier.
+     *
+     * @param clientSupplier        RestHighLevelClient supplier
+     * @param searchRequestSupplier supplier of a SearchRequest used to query for documents
+     * @param mapHitFn              supplier of a function mapping the result from SearchHit to a target type
+     * @param <T>                   result type returned by the map function
+     */
+    public static <T> BatchSource<T> elastic(
+            @Nonnull SupplierEx<RestHighLevelClient> clientSupplier,
+            @Nonnull SupplierEx<SearchRequest> searchRequestSupplier,
```

Comment:
SearchRequest is `Writable` so it could be serialized, rather than requiring a supplier.

Reply:
We would need to create some adapter for org.elasticsearch.common.io.stream.Writeable and the user would need to register it to Jet, right?

Reply:
Yes something like that (there's something similar done in hadoop module) but when supplying it as a parameter, I think you'll need to serialize/deserialize by hand (because parent class is only `Serializable`)

Reply:
So the request can be easily serialized to `byte[]` in following way:

```
ByteArrayOutputStream baos = new ByteArrayOutputStream();
OutputStreamStreamOutput osso = new OutputStreamStreamOutput(baos);
searchRequest.writeTo(osso);
baos.toByteArray();
```
but the deserialization isn't that easy:
```
ByteArrayInputStream bais = new ByteArrayInputStream(searchRequest);
InputStreamStreamInput input = new InputStreamStreamInput(bais);
return new SearchRequest(input);
```
This works for a simple `SearchRequest`, but if some parameters of the SearchRequest are set (e.g. SearchRequest#source) it fails to deserialize, requiring  `NamedWriteableAwareStreamInput` as input, which needs NamedWriteableRegistry - this is an internal elastic class, initialized e.g. here https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/node/Node.java#L397

Also this part of API is different between elastic 5, 6 and 7.

With all this I think it's better to keep the supplier.